### DISCUSSION TITLE
401 Unauthorized via HTTP API

### DISCUSSION THREAD

**Discussion by jordancrubin** [deta/cloud-docs#728] [Help]
This was also tried in postman, modded for security purposes, login data isnt real. It works on my esp32 but not from http not postman.

URL: https://database.deta.sh/v1/a7xy2v6n/esp32project/query

Content-type = application/json
x-api-key = a7xy2v6n_ldjkhdslkdshsdkljhsdlfkjh

BODY:
{"user.age?lt": 40}

RESP:
{
    "errors": [
        "Unauthorized"
    ]
}

**Comment by jnsougata** [comment]
headers are case sensitive in here so try with `X-API-Key`

**Reply by jordancrubin** [reply]
this did not work

X-API-Key

**Reply by jnsougata** [reply]
are you sure it's a project key and not an access token?

**Reply by jordancrubin** [reply]
So strangely the first project key on my database worked on my ESP32. Having created another key, they new one works on both the esp32 and http api, though I cant say why. thanks for the help though its solved now.


<img width="1374" alt="Screenshot 2023-01-20 at 9 12 39 PM" src="https://user-images.githubusercontent.com/12800500/213839485-4963667a-b26c-46dd-9c27-f4266cca4c79.png">


<img width="556" alt="Screenshot 2023-01-20 at 9 10 30 PM" src="https://user-images.githubusercontent.com/12800500/213839133-7135c6a2-1bd5-433a-bd0b-7ee85c45d797.png">

**Reply by jnsougata** [reply]
glad your issue has been fixed!
